### ISSUE TITLE
Devices and Settings shows "iPhone 15" after logging in

### ISSUE THREAD

**Issue by Erulezz** [johannes-schliephake/nextcloud-passwords-ios#191]
Hi,

First of all, thanks for this app and open source it. Great stuff 👍

After logging in with my account I see the following items in the Devices & sessions list:

![Schermafbeelding 2022-05-04 om 22 45 24](https://user-images.githubusercontent.com/6772197/166823130-a04e4c24-9b99-4aa0-b642-8f6d729f641a.png)

I double checked it by revoking these items and logging back in in the app, and again iPhone 15 shows up. Now my question since the iPhone 15 doesn't even exists at this time: why is this showing up? 👀 This is not the name of my device. Other apps show: "iPhone 11 (Nextcloud Talk)"

Using NextCloud 23.0.4.

**Comment by johannes-schliephake** [comment]
Hey! Thanks for your words :)

This is nothing to worry about. It's just the weird way Nextcloud implements the login flow.
What is happening is the iOS app shows you the browser login. After logging in it will ask you to grant access. At this point you will get the "iPhone 15" entry because you technically logged into Nextcloud with "15" actually meaning iOS 15... And after granting access the actual access token is given to the app. The app itself uses the device name "Passwords (iOS)". So if you want to you can actually delete the "iPhone 15" device after granting access if you want to and the app will stay connected. And finally the "iPhone 15" naming is generated by Nextcloud, so unfortunately I can't change the name.

So in summary: Nextcloud is doing weird stuff when loggin in via an app, but everything is still secure.

**Comment by Erulezz** [comment]
Ah now it makes sense, thanks! I deleted the iPhone 15 entry and everything still works 👍

I'm just getting started with Nextcloud so I don't know if this is possible: can the app delete this entry automatically after logging in? The login flow is exactly the same (it looks that way) with the Nextcloud app (and Talk). If that's not possible no big deal, now it's clear where it's coming from.

Also: I can provide Dutch translations, do you want that?

**Comment by johannes-schliephake** [comment]
You are right, the official apps only show up with a single entry. I never noticed that. I guess then it's worth taking a closer look how those apps do the login flow.

A Dutch translation would be awesome! I just finished integrating with Weblate (A translation tool, so translator don't have to use Git). If you want to you can already use it here: https://hosted.weblate.org/engage/nextcloud-passwords-ios/. You can also write me at mail@cdslash.de if you need help or if Weblate needs some additional configuration.

**Comment by Erulezz** [comment]
Nice! I've started with the translation.

**Comment by johannes-schliephake** [comment]
The app will now delete app passwords that are no longer in use. The first session is removed after logging in from the app and the second one after logging out.

I've also taken a look at adding the device name to the app password, but this is actually way to easy to mess up. Special characters are not allowed to occur in the app password's name, so if the user decides to name their device for example "Björn", Nextcloud will break in unpredictable ways.

**Comment by Erulezz** [comment]
Nice 👍🏻 

Translations are done for the most part, if I can find the time / mood I'll finish it

**Comment by johannes-schliephake** [comment]
Awesome! Take your time :)
